src/Tasks/Microsoft.NET.Build.Tasks/RunReadyToRunCompiler.cs

```diff
@@ -208,6 +278,7 @@ private string GenerateCrossgen2ResponseFile()
             }
 
             result.AppendLine("-O");
+            result.AppendLine("--pdb");
```

Comment:
We cannot safely pass --pdb to versions of crossgen2 before 6.0.

Some of the logic above abstracts the differences between 5.0 and 6.0 crossgen2 by checking the jitpath variable, and if it is set, but I think we should regularize this so there is a crossgen2 version that we are speaking of. That would allow us to be more intentional, such as specifying that if working with 5.0, when building pdbs, build them with crossgen.exe instead of crossgen2. When we get 6.0 we can use the `--pdb` switch. If we have a rigourous model where it is clear what code effects which versions of the compiler, I think it will be much easier to get this all correct.

Reply:
Based on your feedback I have modified the change so that --pdb and other "6.0 Crossgen2"-specific features are only triggered when JitPath is not specified. I have managed to validate this change locally after applying Steve's and Anirudh's fixes for rolling forward the runtime SDK drop to 6.0.

Reply:
Note that we do have Crossgen2 version number and it would be trivial to pass it through metadata: https://github.com/dotnet/sdk/blob/419b69015e49347ad9c5d1cc01e28bbd7c140394/src/Tasks/Microsoft.NET.Build.Tasks/ResolveReadyToRunCompilers.cs#L143